source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -104,6 +104,191 @@ TEST(XnetConvertersTests, SessionInfoStateValue_SetSessionInfoResponse_ExtractSe
   EXPECT_EQ(nixnet_grpc::SessionInfoState::SESSION_INFO_STATE_STARTED, output.info());
   EXPECT_EQ(sessionInfoState, output.info_raw());
 }
+
+void assert_frames_are_equal(nxFrameVar_t* frame1, nixnet_grpc::FrameResponse frame2)
+{
+  EXPECT_EQ(frame1->Timestamp, frame2.timestamp());
+  EXPECT_EQ(frame1->Identifier, frame2.identifier());
+  EXPECT_EQ(nixnet_grpc::FrameType::FRAME_TYPE_LIN_DATA, frame2.type());
+  EXPECT_EQ(frame1->Type, frame2.type_raw());
+  EXPECT_EQ(1, frame2.flags().size());
+  EXPECT_EQ(nixnet_grpc::FrameFlags::FRAME_FLAGS_LIN_EVENT_SLOT, frame2.flags()[0]);
+  EXPECT_EQ(frame1->Flags, frame2.flags_raw());
+  EXPECT_EQ(frame1->Info, frame2.info());
+  ASSERT_THAT(std::vector<u8>(frame1->Payload, frame1->Payload + frame1->PayloadLength), ::testing::ElementsAreArray(frame2.payload()));
```

Comment:
Can you also add assert for `frame1->PayloadLength` equals `frame2.payload().size()`

Reply:
Done.